libraries/chain/controller.cpp

```diff
@@ -190,6 +190,11 @@ struct controller_impl {
    map< account_name, map<handler_key, apply_handler> >   apply_handlers;
    unordered_map< builtin_protocol_feature_t, std::function<void(controller_impl&)>, enum_hash<builtin_protocol_feature_t> > protocol_feature_activation_handlers;
 
+   // mutex for account object
+   // this is necessary to ensure thread safety between chain_plugin "get_all_accounts" RPC call 
+   // and account creation, which should be mutually exclusive
+   std::mutex acct_obj_mutex; 
```

Comment:
Let's not do this. No reason for the `get_all_accounts` to be off the main thread. It should just be implemented like the other chain api calls.

Reply:
Done